scripts/hf_sim.py

```diff
@@ -155,11 +158,14 @@ def random_seed():
         default=1,
     )
     try:
-        args = parser.parse_known_args()[0]
+        args = parser.parse_args()
     except SystemExit:
         # invalid arguments or -h
         comm.Abort()
 
+if args.version is not None:
+    args.sim_bin = binary_version.get_hf_binmod(args.version)
```

Comment:
Confusing behaviour

Reply:
i can probably remove this line.
this changes was to band-aid fix when the submit_* code did not handle the version provided in gmsim template correctly.

Reply:
if the submit code, either auto_submit or submit_*. did not handle the 'version' parameter correctly, with the code here it will still run with the version provided.
instead of just using default version

Reply:
Yeah the binary to use is determined in submit_hf.py, so I think this should be removed. And in extension the version argument, since its not used anywhere else?